test/e2e/gardener/shoot/internal/highavailability/highavailability.go

```diff
@@ -0,0 +1,119 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package highavailability
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	gomegatypes "github.com/onsi/gomega/types"
+	istionetworkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	. "github.com/gardener/gardener/test/e2e/gardener"
+)
+
+// VerifyHighAvailabilityUpdate verifies the high-availability update of a shoot cluster. It checks the topology
+// spread constraints, ETCD affinity, and envoy filters to ensure the shoot cluster is correctly upgraded to
+// high-availability configuration.
+func VerifyHighAvailabilityUpdate(s *ShootContext) {
```

Comment:
Nit: this function doesn't verify any kind of "update". It rather verifies the resulting HA configuration.
```suggestion
func VerifyHighAvailability(s *ShootContext) {
```